### DISCUSSION TITLE
Very slow work cron job

### DISCUSSION THREAD

**Discussion by DobroFenix** [nilsteampassnet/TeamPass#4543] [Q&A]
How can I speed up the tasks of creating new elements and users? I have been creating a new user for over an hour, and new items for about 15 minutes. At the same time, there is no load on the server, php is not running in the processes.
The crontab is running once a minute.

Settings tasks:
![image](https://github.com/user-attachments/assets/a91dd6d5-6d73-47fb-8b52-262d1e579a2e)

Completed tasks:
![image](https://github.com/user-attachments/assets/27a1950d-b151-4363-a7a6-3ccba4626411)

**Comment by DobroFenix** [comment]
![image](https://github.com/user-attachments/assets/e776a436-b4b5-4eeb-a0ca-b1bf775fbe54)
Create new item - 2min = slow 
I don't see any complicated calculations when creating a new element. How to do it instantly?
![image](https://github.com/user-attachments/assets/9efdced7-eff9-48cb-9956-f2ae4db17950)

**Comment by corentin-soriano** [comment]
@DobroFenix this job creates the "share keys" of the object so that users can access it.
The object is created as soon as you click the save button, however users can only see the password once they have their share key.
For each item, a key is created per user, which can take a little time depending on the number of users in your database. Note that you must wait the next minute for the scheduler to launch the asynchronous job.

**Reply by DobroFenix** [reply]
Hi!
I want to implement TeamPass in our organization and so far I have 6,000 users who are not currently using the system, but are in the TeamPass database.
Can this be accelerated somehow?

Please help me solve the problem from the adjacent question related to the fact that when creating a new element, I do not have rights to this element :(

**Reply by corentin-soriano** [reply]
With a base of 600 users, I am also at 02:01.
I think that what limits the most is mainly the wait for the cron to launch the rest of the job (there are several steps which are not launched during the same execution.

Maybe this cron could improve the times slightly but I haven't tried.
`* * * * * /usr/bin/php /var/www/teampass/sources/scheduler.php && sleep 30 && /usr/bin/php /var/www/teampass/sources/scheduler.php`
